### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by NatalliaSavitskaya**
1. Attached JUnit and Mockito dependencies to the project.
2. ClientServiceTest class is created.
3. @Test methods were created to cover positive, negative, corner case scenario for each methos from ClientService class.
4. Junit annotations and method were used to „init“ test data and make assertions and verification checks.
5. Mockito was used to mock Database repositiry behaviour.

### CODE REVIEW

File: src/main/java/myApp/service/ClientService.java [hunk 1]
```diff
@@ -23,13 +24,27 @@ public ClientService(ClientRepository clientRepository, OrderRepository orderRep
...
 
+    public void deleteClientById(int clientId) {
+        Client client = clientRepository.findById(clientId).orElseThrow(() ->
+                new IllegalArgumentException("Client not found with ID: " + clientId)
+        );
+        List<Order> orders = orderRepository.findByClientId(clientId);
+        if (!orders.isEmpty()) {
+            throw new UnsupportedOperationException("Client with ID " + clientId + " cannot be deleted because they have associated orders.");
+        }
+        clientRepository.deleteById(clientId);
```
Comment (VolgaDaniluk) [inline src/main/java/myApp/service/ClientService.java:45]: from spring documentation
method in the repository

image

File: src/test/java/myApp/service/ClientServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    @Captor
+    private ArgumentCaptor<Client> clientCaptor;
+
+    @Captor
+    private ArgumentCaptor<Order> orderCaptor;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
```
Comment (VolgaDaniluk) [inline src/test/java/myApp/service/ClientServiceTest.java:41]: if you use such approach you have to add method with @AfterEach to close the resource 
or

simplify the way of mock creation and injection (use @ExtendWith(MockitoExtension.class) as a class annotation and remove setUp method)

File: src/test/java/myApp/service/ClientServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    @Test
+    void testSaveClient_ClientNameIsEmpty() {
+        Client client = new Client(1, "", LocalDateTime.now(), "PENDING");
+        when(clientRepository.save(any(Client.class))).thenReturn(client);
+
+        clientService.saveClient(client);
+
+        verify(clientRepository, times(1)).save(clientCaptor.capture());
+        assertEquals(client, clientCaptor.getValue());
+    }
```
Comment (VolgaDaniluk) [inline src/test/java/myApp/service/ClientServiceTest.java:73]: you've already test this case in the first test method

File: src/test/java/myApp/service/ClientServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+    @Test
+    void testGetClient_ClientIdIs0() {
+        when(clientRepository.findById(0)).thenReturn(Optional.empty());
+
+        Client actualClient = clientService.getClient(0);
+
+        assertNull(actualClient);
+        verify(clientRepository, times(1)).findById(0);
+    }
```
Comment (VolgaDaniluk) [inline src/test/java/myApp/service/ClientServiceTest.java:154]: already covered
